### PR TITLE
Activity Log: Get display rules from real API endpoint

### PR DESCRIPTION
**Pull request by mattgawarecki**
Resolves `1200412004370260-as-1200706668146547`.

#### Changes proposed in this Pull Request

* Update the data-getter endpoint for Activity Log display rules to match the endpoint in `D65474-code`.
* Update application state logic to accommodate the shape of the API response in `D65474-code`.

#### Testing instructions

For extra context, check out #54900 and #55298.

##### Prerequisites

1. Create or select a connected Jetpack site to use for testing this PR.
2. Add a Backup or Security Tier 1 subscription to your site via the Store Admin.
3. Enable the `activity-log/display-rules` feature flag, either by modifying `development.json`/`jetpack-cloud-development.json` or by adding `flags=activity-log/display-rules` to your browser URL's query string.
4. Open your browser's development tools to the Network panel to monitor XHR requests.

##### New behavior

5. Visit the Activity Log page in Calypso (Blue or Green).
6. Verify you see an outgoing request to `/wpcom/v2/sites/<site id>/rewind/policies`, and that it returns a success response.
7. Verify that if you added a Tier 1 subscription, the `activity_log_limit_days` field has a value of `30`; or, if you selected a Tier 2 subscription, it has a value of `365`.
8. In your browser's console, execute `state.rewind[ state.ui.selectedSiteId ].policies` and examine the result:
  * the `requestStatus` field should equal `success`; and
  * the `activityLogLimitDays` field should match the value of `activity_log_limit_days` from the network response in step 8.
9. Test that on sites with a Tier 1 subscription and more than 30 days of Activity history, you see a notice at the 30-day mark with instructions on how to upgrade your storage (see #54900, #55298).

##### Existing behavior

If you'd like to test for regressions:

10. Remove the Tier 1/Tier 2 subscription from your site. Wait up to 5 minutes for back-end caches to update.
11. Refresh the Activity Log page.
12. Verify you see a successful network request like in step 7, albeit with no data for `activity_log_limit_days`.
13. Verify the application state now reflects an `activityLogLimitDays` value of `null`.
14. Verify there are no unexpected errors in the browser console.

### CODE REVIEW

File: client/components/activity-card-list/visible-days-limit-upsell/index.tsx [hunk 1]
```diff
@@ -64,37 +62,37 @@ const RetentionLimitUpsell: React.FC< OwnProps > = ( { cardClassName } ) => {
...
-				<h3 className="retention-limit-upsell__call-to-action-header">
+		<div className="visible-days-limit-upsell">
+			<div className="visible-days-limit-upsell__next-activity">{ card }</div>
+			<div className="visible-days-limit-upsell__call-to-action">
+				<h3 className="visible-days-limit-upsell__call-to-action-header">
 					{ preventWidows(
 						translate(
-							'Restore backups older than %(retentionDays)d day',
-							'Restore backups older than %(retentionDays)d days',
+							'Restore backups older than %(visibleDays)d day',
```
Comment (a8ci18n) [inline client/components/activity-card-list/visible-days-limit-upsell/index.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **15** times:
`translate( 'Restore backups older than %(days)d day', 'Restore backups older than %(days)d days' )` *ES Score: 8*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/activity-log-display-rules-from-api)

File: client/components/activity-card-list/visible-days-limit-upsell/index.tsx [hunk 2]
```diff
@@ -64,37 +62,37 @@ const RetentionLimitUpsell: React.FC< OwnProps > = ( { cardClassName } ) => {
...
 						)
 					) }
 				</h3>
-				<p className="retention-limit-upsell__call-to-action-copy">
+				<p className="visible-days-limit-upsell__call-to-action-copy">
 					{ preventWidows(
 						translate(
-							'Your activity log spans more than %(retentionDays)d day. Upgrade your backup storage to access activity older than %(retentionDays)d day.',
-							'Your activity log spans more than %(retentionDays)d days. Upgrade your backup storage to access activity older than %(retentionDays)d days.',
+							'Your activity log spans more than %(visibleDays)d day. Upgrade your backup storage to access activity older than %(visibleDays)d day.',
```
Comment (a8ci18n) [inline client/components/activity-card-list/visible-days-limit-upsell/index.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **14** times:
`translate( 'Your activity log spans more than %(days)d day. Upgrade your backup storage to access activity older than %(days)d day.', 'Your activity log spans more than %(days)d days. Upgrade your backup storage to access activity older than %(days)d days.' )` *ES Score: 9*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/activity-log-display-rules-from-api)

File: client/state/rewind/selectors/get-activity-log-visible-days.ts [hunk 3]
```diff
@@ -15,7 +15,7 @@ const getActivityLogVisibleDays = ( state: AppState, siteId: number ): number |
 		return undefined;
 	}
 
-	return state.activityLog.displayRules[ siteId as number ]?.visibleDays ?? undefined;
+	return state.rewind?.[ siteId as number ]?.polices?.activityLogLimitDays ?? undefined;
```
Comment (monsieur-z) [inline client/state/rewind/selectors/get-activity-log-visible-days.ts]: There's a typo here: it should be `policies`.

**Review by monsieur-z — CHANGES_REQUESTED** [review]
There's a typo that needs to be fixed (see previous comment). Other than that, in Calypso blue (haven't checked in Cloud), I only see about 20 items before the upsell.

Screen Shot 2021-09-01 at 10 05 51 AM

**Comment by mattgawarecki** [comment]
@monsieur-z

> There's a typo that needs to be fixed (see previous comment).

Fixed! Thanks for catching that, it slipped through during my refactor yesterday.

> Other than that, in Calypso blue (haven't checked in Cloud), I only see about 20 items before the upsell.

I had an inverted conditional that I fixed in 61b2f1f, so this should be resolved now 👍 If your site's Activity Log doesn't span for an entire 30 days, the upsell in Calypso Blue should no longer be visible.

**Comment by mattgawarecki** [comment]
**NOTE:** Yesterday we discovered a gap in the API where new Security and Backup plans aren't having their Activity Log events retained for the correct amount of time. Until this gap is resolved, apply `D66240-code` to a WordPress.com sandbox and alias `public-api.wordpress.com` to point to this sandbox for testing.

**Comment by mattgawarecki** [comment]
Update to my last comment: `D66240-code` has been deployed to production WordPress.com.

**Comment by cbauerman** [comment]
`state.rewind[ state.ui.selectedId ].policies` should be `state.rewind[ state.ui.selectedSiteId ].policies`

**Pull request merged**